### PR TITLE
enable some eslint and typescript rules

### PR DESCRIPTION
**Pull request by xDalete**
this is an optional pr
i did this because i believe these rules will force the code to be more self explanatory and easier to understand 

enabled eslint rules
no-empty-object-type
no-empty-function
no-unused-vars

enabled typescript compiler options
noImplicitAny

removed
ban-types (deprecated)

### CODE REVIEW

File: src/renderer/src/lib/State.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export class State<T extends {}, K extends keyof T = keyof T> {
+export class State<T extends object, K extends keyof T = keyof T> {
```
Comment (brw) [inline src/renderer/src/lib/State.ts:1]: I feel like this should be a record instead but also this class doesn't actually seem to be used anywhere (and I don't see why you'd use it over a regular Map) so might as well delete the file lol

**Review by brw — CHANGES_REQUESTED** [review]

**Comment by brw** [comment]
Other than that lgtm though

**Review by brw — APPROVED** [review]

File: src/@types.d.ts [hunk 2]
```diff
@@ -168,11 +168,20 @@ export type Tag = {
   isSpecial?: boolean;
 };
 
+export type OrderOptions = "title" | "artist" | "creator" | "bpm" | "duration" | "dateAdded";
+
+export type OrderDirection = "asc" | "desc";
+
+export type OrderType = {
```
Comment (brw) [inline src/@types.d.ts:175]: Is there a reason for appending "Type" to this and not just calling it `Order`?

Comment (xDalete) [reply src/@types.d.ts:175]: I just have the habit of adding "Type" or "Props" at the end of names
Do you want me to remove it?

Comment (brw) [reply src/@types.d.ts:175]: yea I'd remove it, it's redundant

**Pull request merged**